site/content/blog/2019-07-18-chain-react-2019.md

```diff
@@ -0,0 +1,123 @@
+the MC a very entertaining dude, opened the way to our first speaker:
+
+#### Improving React Native Performance (Marc Horowitz)
+
+Right from the start the Facebook React Native core team made a very BIG announcement, 
+this was all about improving performance on cold start (app start), 
+taking into consideration three primary metrics: 
+`TTI (Time to Interaction)`, `APK (application size)` and `Memory (as in utilization)`.
+
+So what does this mean?
+
+# 🎉 Hermes welcome!
+
+Hermes is a new JavaScript engine designed for mobile environments, that its focused on 
+startup performance, after this short explanation by Marc, 
+I personally went nuts! right there and without any knowledge of what was about to 
+happen a video pops up in the screen.
+
+[![Mattermost JSC vs Herms](/blog/2019-07-18-chain-react-2019/thumbnail.png)](https://youtu.be/EnefPsVRSn4)
+*<span style="font-size: 12px">click on the image to open the YouTube video.</span>*
+
+BOOM BOOM POW! as you can see in the video above, the RN core team wanted to show the results on a 
+real world app with complex interactions, layouts and functionality and they chose Mattermost. 
+The Mattermost app was shown as an example of what Hermes can achieve, in our app case just by 
+switching from JSC to Hermes it improved cold start in about 50% on a Pixel XL (2016) phone.
+
+Here more detailed results:
+![Mattermost mobile app](/blog/2019-07-18-chain-react-2019/hermes-stats.jpg)
+
+And a few more comparisons thanks to [Ram Narasimhan](blog.nparashuram.com):
+
+   - TTI Traces - [JSC](https://nparashuram.github.io/mattermost-mobile/jsc/trace-jsc.html) vs [Hermes](https://nparashuram.github.io/mattermost-mobile/hermes/trace-hermes.html)
+   - Memory Usage - [JSC](https://nparashuram.github.io/mattermost-mobile/jsc/mem-jsc.txt) vs [Hermes](https://nparashuram.github.io/mattermost-mobile/hermes/mem-hermes.txt)
+   - APK Size - [JSC](https://nparashuram.github.io/mattermost-mobile/jsc/size-jsc.html) vs [Hermes](https://nparashuram.github.io/mattermost-mobile/hermes/size-hermes.html)
+
+
+Ok lets dig a little deeper and see why Hermes improves cold start. When you build a mobile app with 
+React Native basically all your JS code is transpiled with babel to convert your ECMAScript 2015+ 
+code into a backwards compatible version of JavaScript in current and older environments, then the 
+code is minified to reduce its bundle size and the bundle is saved within your app. When you run the 
+app the JavaScript engine kicks in, parses the bundle and compiles it into bytecode and finally it 
+gets executed. Hermes changes thing a bit, your app is still transpiled with babel and then minified, 
+but Hermes uses an ahead-of-time compiler, which runs as part of the mobile app build process, 
+that also allows to spend more time optimizing the bytecode, and with this the bytecode is smaller in 
+size and more efficient. Some of the optimizations that are performed include function deduplication 
```

Comment:
From a technical perspective, I'm wondering if there is anything we can flesh out here in more detail -- deduplicating functions and packing string tables /seems/ like it wouldn't cause a 2x performance boost, but I may not fully understand the impact.

Reply:
There's not much that I can say about this, as I don't really know the impact in performance that we'll get, but this is highlighted by Hermes as some of the optimizations that they perform during build time.

> namely when we think we might ship these improvements, an itemized list of blockers, etc.

We are currently implementing and it would be awesome for @migbot to write another blog post after he finishes with the implementation as a follow up post